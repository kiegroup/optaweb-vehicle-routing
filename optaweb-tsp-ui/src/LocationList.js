import PropTypes from 'prop-types';
import React from 'react';
import Location from './Location';

function LocationList({ route, domicileId, removeHandler, selectHandler }) {
  return (
    <div className={'leaflet-top leaflet-left leaflet-touch'}>
      <div className={'leaflet-control leaflet-bar w5 bg-white '}>
        {
          route.length === 0 ?
            <div className={'tc ma2'}>Click map to add locations</div> :
            route
              .slice(0) // clone the array
              .sort((a, b) => a.id - b.id) // because sort is done in place (that would affect the route)
              .map(location => (
                <Location
                  key={location.id}
                  id={location.id}
                  removeDisabled={route.length > 1 && location.id === domicileId}
                  removeHandler={removeHandler}
                  selectHandler={selectHandler}
                />
              ))
        }
      </div>
    </div>
  );
}

LocationList.propTypes = {
  route: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    lat: PropTypes.number.isRequired,
    lng: PropTypes.number.isRequired,
  })).isRequired,
  domicileId: PropTypes.number.isRequired,
  removeHandler: PropTypes.func.isRequired,
  selectHandler: PropTypes.func.isRequired,
};

export default LocationList;
