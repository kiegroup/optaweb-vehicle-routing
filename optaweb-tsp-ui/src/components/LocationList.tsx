/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Button, Card, CardBody, CardHeader } from '@patternfly/react-core';
import * as React from 'react';
import { ITSPRoute } from '../store/tsp/types';
import Location from './Location';
import TripData from './TripData';

export interface ILocationListProps extends ITSPRoute {
  removeHandler: (id: number) => void;
  selectHandler: (e: any) => void;
  loadHandler: () => void;
  maxDistance: number;
}

const renderEmptyLocationList = ({ loadHandler }: ILocationListProps) => {
  return (
    <Card>
      <CardHeader>Click map to add locations</CardHeader>
      <CardBody>
        <Button type="button" style={{ width: '100%' }} onClick={loadHandler}>
          Load 40 European cities
        </Button>
      </CardBody>
    </Card>
  );
};

const renderLocationList = ({
  route,
  domicileId,
  distance = '',
  removeHandler,
  selectHandler,
  loadHandler,
  maxDistance,
}: ILocationListProps) => {
  return (
    <Card>
      <CardHeader>
        Distance: {distance}
        <br />
        Locations: {route.length}
        <TripData
          maxDistance={maxDistance}
          distance={parseInt(distance, 10) || maxDistance}
        />
        <hr />
      </CardHeader>
      <CardBody>
        {/*
               The calculated maxHeight is a hack because the constant 116px depends
               on the height of Distance and Locations rows (above) and individual location rows.
               */}
        <div style={{ maxHeight: 'calc(100vh - 195px)', overflowY: 'auto' }}>
          {route
            .slice(0) // clone the array because
            // sort is done in place (that would affect the route)
            .sort((a, b) => a.id - b.id)
            .map(location => (
              <Location
                key={location.id}
                id={location.id}
                removeDisabled={route.length > 1 && location.id === domicileId}
                removeHandler={removeHandler}
                selectHandler={selectHandler}
              />
            ))}
        </div>
      </CardBody>
    </Card>
  );
};

const LocationList: React.SFC<ILocationListProps> = (
  props: ILocationListProps
) => {
  return (
    <div
      className="leaflet-top leaflet-left leaflet-touch"
      style={{ zIndex: 500 }}
    >
      <div className="leaflet-control leaflet-bar">
        {props.route.length === 0
          ? renderEmptyLocationList(props)
          : renderLocationList(props)}
      </div>
    </div>
  );
};

export default LocationList;
