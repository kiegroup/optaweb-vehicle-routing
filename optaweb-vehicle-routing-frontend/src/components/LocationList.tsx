/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Button, Card, CardBody, CardHeader } from '@patternfly/react-core';
import * as React from 'react';
import { IRoute } from '../store/route/types';
import Location from './Location';
import SearchBox from './SearchBox';
import TripData from './TripData';

export interface ILocationListProps {
  removeHandler: (id: number) => void;
  selectHandler: (id: number) => void;
  loadHandler: () => void;
  clearHandler: () => void;
  maxDistance: number;
  route: IRoute;
  domicileId: number;
  isDemoLoading: boolean;
}

const renderEmptyLocationList = ({
  loadHandler,
  isDemoLoading,
}: ILocationListProps) => {
  return (
    <Card>
      <CardHeader>Click map to add locations</CardHeader>
      <CardBody>
        <Button
          type="button"
          isDisabled={isDemoLoading}
          style={{ width: '100%' }}
          onClick={loadHandler}
        >
          Load demo
        </Button>
      </CardBody>
    </Card>
  );
};

const renderLocationList = ({
  route: { distance, locations },
  domicileId,
  removeHandler,
  selectHandler,
  clearHandler,
  maxDistance,
  isDemoLoading,
}: ILocationListProps) => {
  return (
    <Card>
      <CardHeader>
        Distance: {distance}
        <br />
        Locations: {locations.length}
        <TripData
          maxDistance={maxDistance}
          distance={parseInt(distance, 10) || maxDistance}
        />
        <br />
        <Button
          type="button"
          isDisabled={isDemoLoading}
          style={{ width: '100%' }}
          onClick={clearHandler}
        >
          Clear
        </Button>
        <br />
        <SearchBox />
      </CardHeader>
      <CardBody>
        {/*
        The calculated maxHeight is a hack because the last constant depends
        on the height of CardHeader (above).
        */}
        <div style={{ maxHeight: 'calc(100vh - 24px - 24px - 8px - 196px)', overflowY: 'auto' }}>
          {locations
            .slice(0) // clone the array because
            // sort is done in place (that would affect the route)
            .sort((a, b) => a.id - b.id)
            .map(location => (
              <Location
                key={location.id}
                id={location.id}
                removeDisabled={locations.length > 1 && location.id === domicileId}
                removeHandler={removeHandler}
                selectHandler={selectHandler}
              />
            ))}
        </div>
      </CardBody>
    </Card>
  );
};

const LocationList: React.SFC<ILocationListProps> = (
  props: ILocationListProps,
) => {
  return (
    <div
      className="leaflet-top leaflet-left leaflet-touch"
      style={{ zIndex: 500 }}
    >
      <div className="leaflet-control leaflet-bar">
        {props.route.locations.length === 0
          ? renderEmptyLocationList(props)
          : renderLocationList(props)}
      </div>
    </div>
  );
};

export default LocationList;
