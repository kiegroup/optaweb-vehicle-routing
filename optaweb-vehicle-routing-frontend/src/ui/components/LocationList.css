.pf-c-data-list__cell {
  --pf-c-data-list__cell-cell--PaddingTop: var(--pf-global--spacer--sm);
  --pf-c-data-list__cell--PaddingTop: var(--pf-global--spacer--md);
  --pf-c-data-list__cell--PaddingBottom: var(--pf-global--spacer--sm);
}
