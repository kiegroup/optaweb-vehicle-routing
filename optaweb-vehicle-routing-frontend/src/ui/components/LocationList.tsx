/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Bullseye, DataList } from '@patternfly/react-core';
import * as React from 'react';
import { Location } from 'store/route/types';
import LocationItem from './Location';
import './LocationList.css';

export interface LocationListProps {
  removeHandler: (id: number) => void;
  selectHandler: (id: number) => void;
  depot: Location | null;
  visits: Location[];
}

const renderEmptyLocationList: React.FC<LocationListProps> = () => (
  <DataList aria-label="Empty location list">
    <Bullseye>No locations</Bullseye>
  </DataList>
);

const renderLocationList: React.FC<LocationListProps> = ({
  depot,
  visits,
  removeHandler,
  selectHandler,
}) => (
  <div style={{ overflowY: 'auto' }}>
    <DataList
      aria-label="List of locations"
    >
      {depot && (
        <LocationItem
          key={depot.id}
          id={depot.id}
          description={depot.description || null}
          removeDisabled={visits.length > 0}
          removeHandler={removeHandler}
          selectHandler={selectHandler}
        />
      )}
      {visits
        .slice(0) // clone the array because
        // sort is done in place (that would affect the route)
        .sort((a, b) => a.id - b.id)
        .map(visit => (
          <LocationItem
            key={visit.id}
            id={visit.id}
            description={visit.description || null}
            removeDisabled={false}
            removeHandler={removeHandler}
            selectHandler={selectHandler}
          />
        ))}
    </DataList>
  </div>
);

const LocationList: React.FC<LocationListProps> = props => (
  props.visits.length === 0 && props.depot === null
    ? renderEmptyLocationList(props)
    : renderLocationList(props)
);

export default LocationList;
