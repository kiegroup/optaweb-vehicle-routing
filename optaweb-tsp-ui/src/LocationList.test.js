/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import LocationList from './LocationList';

describe('Location List Component Test', () => {
  it('should render correctly with no routes', () => {
    const mockLocationList = {
      route: [],
      domicileId: 1,
      distance: '10',
      removeHandler: jest.fn(() => {}),
      selectHandler: jest.fn(select => select),
      loadHandler: jest.fn(() => {}),
    };
    expect.assertions(2);
    const LocationListMock = shallow(<LocationList {...mockLocationList} />);
    expect(LocationListMock).toMatchSnapshot();

    LocationListMock.find('button').simulate('click');

    expect(mockLocationList.loadHandler.mock.calls.length).toBe(1);
  });

  it('should render correctly with few routes', () => {
    const mockLocationList = {
      route: [
        {
          id: 1,
          lat: 1.345678,
          lng: 1.345678,
        },
        {
          id: 2,
          lat: 2.345678,
          lng: 2.345678,
        },
        {
          id: 3,
          lat: 3.676111,
          lng: 3.568333,
        },
      ],
      domicileId: 1,
      distance: '10',
      removeHandler: jest.fn(() => {}),
      selectHandler: jest.fn(select => select),
      loadHandler: jest.fn(() => {}),
    };
    expect.assertions(2);
    const LocationMock = shallow(<LocationList {...mockLocationList} />);
    expect(LocationMock).toMatchSnapshot();

    expect(LocationMock.find('Location').length).toBe(mockLocationList.route.length);

    
  });
  
});
