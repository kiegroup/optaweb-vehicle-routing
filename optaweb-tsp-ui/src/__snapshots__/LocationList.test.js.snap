// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location List Component Test should render correctly with few routes 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocationList
    distance="10"
    domicileId={1}
    loadHandler={[Function]}
    removeHandler={[Function]}
    route={
        Array [
            Object {
              "id": 1,
              "lat": 1.345678,
              "lng": 1.345678,
            },
            Object {
              "id": 2,
              "lat": 2.345678,
              "lng": 2.345678,
            },
            Object {
              "id": 3,
              "lat": 3.676111,
              "lng": 3.568333,
            },
          ]
    }
    selectHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="leaflet-control leaflet-bar w5 bg-white"
>
        <div>
                <div
                        className="tl ma2 pa2"
                >
                        Distance: 
                        10
                </div>
                <div
                        className="tl ma2 pa2"
                >
                        Locations: 
                        3
                </div>
                <div
                        style={
                                Object {
                                        "maxHeight": "calc(100vh - 116px)",
                                        "overflowY": "auto",
                                      }
                        }
                >
                        <Location
                                id={1}
                                removeDisabled={true}
                                removeHandler={[Function]}
                                selectHandler={[Function]}
                        />
                        <Location
                                id={2}
                                removeDisabled={false}
                                removeHandler={[Function]}
                                selectHandler={[Function]}
                        />
                        <Location
                                id={3}
                                removeDisabled={false}
                                removeHandler={[Function]}
                                selectHandler={[Function]}
                        />
                </div>
        </div>
</div>,
      "className": "leaflet-top leaflet-left leaflet-touch",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    className="tl ma2 pa2"
          >
                    Distance: 
                    10
          </div>
          <div
                    className="tl ma2 pa2"
          >
                    Locations: 
                    3
          </div>
          <div
                    style={
                              Object {
                                        "maxHeight": "calc(100vh - 116px)",
                                        "overflowY": "auto",
                                      }
                    }
          >
                    <Location
                              id={1}
                              removeDisabled={true}
                              removeHandler={[Function]}
                              selectHandler={[Function]}
                    />
                    <Location
                              id={2}
                              removeDisabled={false}
                              removeHandler={[Function]}
                              selectHandler={[Function]}
                    />
                    <Location
                              id={3}
                              removeDisabled={false}
                              removeHandler={[Function]}
                              selectHandler={[Function]}
                    />
          </div>
</div>,
        "className": "leaflet-control leaflet-bar w5 bg-white",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="tl ma2 pa2"
>
              Distance: 
              10
</div>,
            <div
              className="tl ma2 pa2"
>
              Locations: 
              3
</div>,
            <div
              style={
                            Object {
                                          "maxHeight": "calc(100vh - 116px)",
                                          "overflowY": "auto",
                                        }
              }
>
              <Location
                            id={1}
                            removeDisabled={true}
                            removeHandler={[Function]}
                            selectHandler={[Function]}
              />
              <Location
                            id={2}
                            removeDisabled={false}
                            removeHandler={[Function]}
                            selectHandler={[Function]}
              />
              <Location
                            id={3}
                            removeDisabled={false}
                            removeHandler={[Function]}
                            selectHandler={[Function]}
              />
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Distance: ",
                "10",
              ],
              "className": "tl ma2 pa2",
            },
            "ref": null,
            "rendered": Array [
              "Distance: ",
              "10",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Locations: ",
                3,
              ],
              "className": "tl ma2 pa2",
            },
            "ref": null,
            "rendered": Array [
              "Locations: ",
              3,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Location
                  id={1}
                  removeDisabled={true}
                  removeHandler={[Function]}
                  selectHandler={[Function]}
/>,
                <Location
                  id={2}
                  removeDisabled={false}
                  removeHandler={[Function]}
                  selectHandler={[Function]}
/>,
                <Location
                  id={3}
                  removeDisabled={false}
                  removeHandler={[Function]}
                  selectHandler={[Function]}
/>,
              ],
              "style": Object {
                "maxHeight": "calc(100vh - 116px)",
                "overflowY": "auto",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "id": 1,
                  "removeDisabled": true,
                  "removeHandler": [Function],
                  "selectHandler": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "function",
                "props": Object {
                  "id": 2,
                  "removeDisabled": false,
                  "removeHandler": [Function],
                  "selectHandler": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "function",
                "props": Object {
                  "id": 3,
                  "removeDisabled": false,
                  "removeHandler": [Function],
                  "selectHandler": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="leaflet-control leaflet-bar w5 bg-white"
>
          <div>
                    <div
                              className="tl ma2 pa2"
                    >
                              Distance: 
                              10
                    </div>
                    <div
                              className="tl ma2 pa2"
                    >
                              Locations: 
                              3
                    </div>
                    <div
                              style={
                                        Object {
                                                  "maxHeight": "calc(100vh - 116px)",
                                                  "overflowY": "auto",
                                                }
                              }
                    >
                              <Location
                                        id={1}
                                        removeDisabled={true}
                                        removeHandler={[Function]}
                                        selectHandler={[Function]}
                              />
                              <Location
                                        id={2}
                                        removeDisabled={false}
                                        removeHandler={[Function]}
                                        selectHandler={[Function]}
                              />
                              <Location
                                        id={3}
                                        removeDisabled={false}
                                        removeHandler={[Function]}
                                        selectHandler={[Function]}
                              />
                    </div>
          </div>
</div>,
        "className": "leaflet-top leaflet-left leaflet-touch",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <div
                        className="tl ma2 pa2"
            >
                        Distance: 
                        10
            </div>
            <div
                        className="tl ma2 pa2"
            >
                        Locations: 
                        3
            </div>
            <div
                        style={
                                    Object {
                                                "maxHeight": "calc(100vh - 116px)",
                                                "overflowY": "auto",
                                              }
                        }
            >
                        <Location
                                    id={1}
                                    removeDisabled={true}
                                    removeHandler={[Function]}
                                    selectHandler={[Function]}
                        />
                        <Location
                                    id={2}
                                    removeDisabled={false}
                                    removeHandler={[Function]}
                                    selectHandler={[Function]}
                        />
                        <Location
                                    id={3}
                                    removeDisabled={false}
                                    removeHandler={[Function]}
                                    selectHandler={[Function]}
                        />
            </div>
</div>,
          "className": "leaflet-control leaflet-bar w5 bg-white",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="tl ma2 pa2"
>
                Distance: 
                10
</div>,
              <div
                className="tl ma2 pa2"
>
                Locations: 
                3
</div>,
              <div
                style={
                                Object {
                                                "maxHeight": "calc(100vh - 116px)",
                                                "overflowY": "auto",
                                              }
                }
>
                <Location
                                id={1}
                                removeDisabled={true}
                                removeHandler={[Function]}
                                selectHandler={[Function]}
                />
                <Location
                                id={2}
                                removeDisabled={false}
                                removeHandler={[Function]}
                                selectHandler={[Function]}
                />
                <Location
                                id={3}
                                removeDisabled={false}
                                removeHandler={[Function]}
                                selectHandler={[Function]}
                />
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Distance: ",
                  "10",
                ],
                "className": "tl ma2 pa2",
              },
              "ref": null,
              "rendered": Array [
                "Distance: ",
                "10",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Locations: ",
                  3,
                ],
                "className": "tl ma2 pa2",
              },
              "ref": null,
              "rendered": Array [
                "Locations: ",
                3,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Location
                    id={1}
                    removeDisabled={true}
                    removeHandler={[Function]}
                    selectHandler={[Function]}
/>,
                  <Location
                    id={2}
                    removeDisabled={false}
                    removeHandler={[Function]}
                    selectHandler={[Function]}
/>,
                  <Location
                    id={3}
                    removeDisabled={false}
                    removeHandler={[Function]}
                    selectHandler={[Function]}
/>,
                ],
                "style": Object {
                  "maxHeight": "calc(100vh - 116px)",
                  "overflowY": "auto",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "function",
                  "props": Object {
                    "id": 1,
                    "removeDisabled": true,
                    "removeHandler": [Function],
                    "selectHandler": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "function",
                  "props": Object {
                    "id": 2,
                    "removeDisabled": false,
                    "removeHandler": [Function],
                    "selectHandler": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "function",
                  "props": Object {
                    "id": 3,
                    "removeDisabled": false,
                    "removeHandler": [Function],
                    "selectHandler": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Location List Component Test should render correctly with no routes 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocationList
    distance="10"
    domicileId={1}
    loadHandler={[Function]}
    removeHandler={[Function]}
    route={Array []}
    selectHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="leaflet-control leaflet-bar w5 bg-white"
>
        <div
                className="tc ma2"
        >
                <div>
                        Click map to add locations
                </div>
                <div>
                        or 
                        <button
                                onClick={[Function]}
                                style={
                                        Object {
                                                "width": "100%",
                                              }
                                }
                        >
                                Load 40 European cities
                        </button>
                </div>
        </div>
</div>,
      "className": "leaflet-top leaflet-left leaflet-touch",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="tc ma2"
>
          <div>
                    Click map to add locations
          </div>
          <div>
                    or 
                    <button
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "width": "100%",
                                                }
                              }
                    >
                              Load 40 European cities
                    </button>
          </div>
</div>,
        "className": "leaflet-control leaflet-bar w5 bg-white",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Click map to add locations
</div>,
            <div>
              or 
              <button
                            onClick={[Function]}
                            style={
                                          Object {
                                                        "width": "100%",
                                                      }
                            }
              >
                            Load 40 European cities
              </button>
</div>,
          ],
          "className": "tc ma2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Click map to add locations",
            },
            "ref": null,
            "rendered": "Click map to add locations",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "or ",
                <button
                  onClick={[Function]}
                  style={
                                    Object {
                                                      "width": "100%",
                                                    }
                  }
>
                  Load 40 European cities
</button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "or ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Load 40 European cities",
                  "onClick": [Function],
                  "style": Object {
                    "width": "100%",
                  },
                },
                "ref": null,
                "rendered": "Load 40 European cities",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="leaflet-control leaflet-bar w5 bg-white"
>
          <div
                    className="tc ma2"
          >
                    <div>
                              Click map to add locations
                    </div>
                    <div>
                              or 
                              <button
                                        onClick={[Function]}
                                        style={
                                                  Object {
                                                            "width": "100%",
                                                          }
                                        }
                              >
                                        Load 40 European cities
                              </button>
                    </div>
          </div>
</div>,
        "className": "leaflet-top leaflet-left leaflet-touch",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="tc ma2"
>
            <div>
                        Click map to add locations
            </div>
            <div>
                        or 
                        <button
                                    onClick={[Function]}
                                    style={
                                                Object {
                                                            "width": "100%",
                                                          }
                                    }
                        >
                                    Load 40 European cities
                        </button>
            </div>
</div>,
          "className": "leaflet-control leaflet-bar w5 bg-white",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Click map to add locations
</div>,
              <div>
                or 
                <button
                                onClick={[Function]}
                                style={
                                                Object {
                                                                "width": "100%",
                                                              }
                                }
                >
                                Load 40 European cities
                </button>
</div>,
            ],
            "className": "tc ma2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Click map to add locations",
              },
              "ref": null,
              "rendered": "Click map to add locations",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "or ",
                  <button
                    onClick={[Function]}
                    style={
                                        Object {
                                                            "width": "100%",
                                                          }
                    }
>
                    Load 40 European cities
</button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "or ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Load 40 European cities",
                    "onClick": [Function],
                    "style": Object {
                      "width": "100%",
                    },
                  },
                  "ref": null,
                  "rendered": "Load 40 European cities",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
